go-controller/pkg/ovn/obj_retry.go

```diff
@@ -0,0 +1,139 @@
+package ovn
+
+import (
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+const retryObjInterval = 30 * time.Second
+
+// retryObjEntry is a generic object caching with retry mechanism
+//that resources can use to eventually complete their intended operations.
+type retryObjEntry struct {
+	newObj     interface{}
+	oldObj     interface{}
+	config     interface{}
+	timeStamp  time.Time
+	backoffSec time.Duration
+	// whether to include this object in the retry iterations
+	ignore bool
+}
+
+// initRetryObjWithAdd tracks an object that failed to be created to potentially retry later
+// initially it is marked as skipped for retry loop (ignore = true)
+func (r *retryObjs) initRetryObjWithAdd(obj interface{}, key string) {
+	r.retryMutex.Lock()
+	defer r.retryMutex.Unlock()
+	if entry, ok := r.entries[key]; ok {
+		entry.timeStamp = time.Now()
+		entry.newObj = obj
+	} else {
+		r.entries[key] = &retryObjEntry{newObj: obj,
+			timeStamp: time.Now(), backoffSec: 1, ignore: true}
+	}
+}
+
+// initRetryWithDelete tracks an object that failed to be deleted to potentially retry later
+// initially it is marked as skipped for retry loop (ignore = true)
+func (r *retryObjs) initRetryObjWithDelete(obj interface{}, key string, config interface{}) {
+	r.retryMutex.Lock()
+	defer r.retryMutex.Unlock()
+	if entry, ok := r.entries[key]; ok {
+		entry.timeStamp = time.Now()
+		entry.oldObj = obj
+		if entry.config == nil {
+			entry.config = config
+		}
+	} else {
+		r.entries[key] = &retryObjEntry{oldObj: obj, config: config,
+			timeStamp: time.Now(), backoffSec: 1, ignore: true}
+	}
+}
+
+// addDeleteToRetryObj adds an old object that needs to be cleaned up to a retry object
+// includes the config object as well in case the namespace is removed and the object is orphaned from
+// the namespace
+func (r *retryObjs) addDeleteToRetryObj(obj interface{}, key string, config interface{}) {
+	r.retryMutex.Lock()
+	defer r.retryMutex.Unlock()
+	if entry, ok := r.entries[key]; ok {
+		entry.oldObj = obj
+		entry.config = config
+	}
+}
+
+// removeDeleteFromRetryObj removes any old object from a retry entry
+func (r *retryObjs) removeDeleteFromRetryObj(key string) {
+	r.retryMutex.Lock()
+	defer r.retryMutex.Unlock()
+	if entry, ok := r.entries[key]; ok {
+		entry.oldObj = nil
+		entry.config = nil
+	}
+}
+
+// unSkipRetryObj ensures an obj is no longer ignored for retry loop
+func (r *retryObjs) unSkipRetryObj(key string) {
+	r.retryMutex.Lock()
+	defer r.retryMutex.Unlock()
+	if entry, ok := r.entries[key]; ok {
+		entry.ignore = false
+	}
+}
+
+// checkAndDeleteRetryObj deletes a specific entry from the map
+func (r *retryObjs) checkAndDeleteRetryObj(key string, withLock bool) {
+	if withLock {
+		r.retryMutex.Lock()
+		defer r.retryMutex.Unlock()
+	}
+	delete(r.entries, key)
+}
+
+// checkAndSkipRetryObj sets a specific entry from the map to be ignored for subsequent retries
+func (r *retryObjs) checkAndSkipRetryObj(key string) {
```

Comment:
do we really need "check"? feels overloaded, since we have `unSkipRetryObj` itsn't `skipRetryObj` enough? its implicit that if the value doesn't exist in the map nothing will be done anyways,

Reply:
I am keeping backward comptability with existing implementation

Reply:
why is that? isn't this the fresh start of a standard set of retryObjs? Isn't it better to be precise with function namings? there is no backwards compatibility with namings...not sure what you mean by "backward compatibility" here...